src/config/mongoose_config_parser_toml.erl

```diff
@@ -194,8 +194,38 @@ http_listener_opt([<<"protocol">>|_] = Path, Opts) ->
     [{protocol_options, parse_section(Path, Opts)}];
 http_listener_opt([<<"handlers">>|_] = Path, Handlers) ->
     [{modules, parse_section(Path, Handlers)}];
+http_listener_opt([<<"mongoose_api_admin">>|_] = Path, V) ->
+    Opts = parse_section(Path, V),
+    Host = proplists:get_value(host, Opts),
+    P = proplists:get_value(path, Opts),
+    User = proplists:get_value(username, Opts),
+    Password = proplists:get_value(password, Opts),
```

Comment:
Quick remark: `proplists:get_value` can return `undefined` and we most likely don't want it. Could you amend all the `proplists:get_value` calls? I'd replcae them with `{_, Value} = proplists:lookup(Key, Opts)`

Reply:
@chrzaszcz if we do `{_, Value} = proplists:lookup(Key, Opts)` and the key wasn't present, `lookup/2` would return `none`, which would fail with a pattern-match. Is that what we want?

Reply:
I can change that, but in this case we get `none` if the key does not exist instead of `undefined` - is it any better?

Reply:
@NelsonVides we want to fail, this is precisely what we want. There is no point in cosntructing a tuple when a mandatory element is not present. This is a broken config and we want to fail fast.